pennylane/transforms/get_dag_commutation.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A transform to obtain the commutation DAG of a quantum circuit.
+"""
```

Comment:
@rmoyard might make sense to move:

- `CommutationDAG` and `CommutationDAGNode` here
- `is_commuting` potentially here as well, but keep it imported top-level?

Reply:
@josh146 I totally agree!